src/NovelAudioService.cpp

```diff
@@ -0,0 +1,367 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#include "NovelAudioService.h"
+#include <iostream>
+#include <functional>
+
+namespace NovelRT {
+
+NovelAudioService::NovelAudioService() : _nextChannel(1), _musicTime(0), _musicPausedTime(0), isInitialized(false) {
+  initializeAudio();
+}
+
+bool NovelAudioService::initializeAudio() {
+  if (SDL_InitSubSystem(SDL_INIT_AUDIO) < 0 )
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL Error: " << SDL_GetError() << std::endl;
+  }
+  else if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 2048) < 0)
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL_Mixer Error: " << Mix_GetError() << std::endl;
+  }
+  else if (Mix_AllocateChannels(2) < 0)
+  {
+    std::cout << "ERROR: Failed to allocate channels." << std::endl;
+  }
+  else
+  {
+    std::cout << "INFO: Mixer engaged!" << std::endl;
+    isInitialized = true;
+  }
+  return isInitialized;
+}
+
+void NovelAudioService::loadSound(std::string input) {
+  auto exists = _sounds.find(input);
+  if (exists != _sounds.end()) return;
+
+  Mix_Chunk* newSound = Mix_LoadWAV(input.c_str());
+  if (newSound != nullptr)
+  {
+    _sounds[input] = newSound;
+  }
+  else
+  {
+    std::cerr << Mix_GetError() << std::endl;
+  }
+
+
+}
+
+void NovelAudioService::unloadSound(std::string input) {
+  auto existingSound = _sounds.find(input);
+  if (existingSound == _sounds.end()) return;
+
+  Mix_FreeChunk(_sounds[input]);
+  _sounds.erase(existingSound);
+}
+
+void NovelAudioService::playSound(std::string soundName) {
+  auto existingSound = _sounds.find(soundName);
+  if (existingSound == _sounds.end())
+  {
+    loadSound(soundName);
+    existingSound = _sounds.find(soundName);
+    if (existingSound == _sounds.end())
+    {
+      return;
+    }
+  }
+
+  auto onChannel = _channelMap.find(soundName);
+  if (onChannel == _channelMap.end())
+  {
+    _channelMap[soundName] = _nextChannel;
+    Mix_PlayChannel(_nextChannel, _sounds[soundName], 0);
+    int nextChannelTest = _nextChannel + 1;
+    _nextChannel = (nextChannelTest >= 2 || nextChannelTest < 0) ? 0 : nextChannelTest;
```

Comment:
You only support stereo audio right now?

Reply:
SDL_Mixer by default only supports stereo audio - currently, I do define to use only two mixing channels.

Reply:
Might be good to define this as a field/property; so it can be easily changed in the future (and it isn't a hunt and search for everywhere you hardcoded `2`).

Reply:
Most recent push actually addresses this by putting in a `const` in the header so it can be reassigned if need be.